crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -143,93 +297,201 @@ impl<F, T>
+        };
+        Ok(Self {
+            response,
+            ..item.data
+        })
+    }
 }
 
-impl From<RefundStatus> for enums::RefundStatus {
-    fn from(item: RefundStatus) -> Self {
-        match item {
-            RefundStatus::Succeeded => Self::Success,
-            RefundStatus::Failed => Self::Failure,
-            RefundStatus::Processing => Self::Pending,
-            //TODO: Review mapping
+impl TryFrom<types::PaymentsSyncResponseRouterData<DataTransSyncResponse>>
+    for types::PaymentsSyncRouterData
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::PaymentsSyncResponseRouterData<DataTransSyncResponse>,
+    ) -> Result<Self, Self::Error> {
+        match item.response {
+            DataTransSyncResponse::Error(error) => {
+                let response = Err(types::ErrorResponse {
+                    code: error.code.unwrap_or(consts::NO_ERROR_CODE.to_string()),
+                    message: error
+                        .message
+                        .unwrap_or(consts::NO_ERROR_MESSAGE.to_string()),
+                    reason: None,
+                    attempt_status: None,
+                    connector_transaction_id: None,
+                    status_code: item.http_code,
+                });
+                Ok(Self {
+                    response,
+                    ..item.data
+                })
+            }
+            DataTransSyncResponse::Response(response) => {
+                let resource_id =
+                    types::ResponseId::ConnectorTransactionId(response.transaction_id.to_string());
+                Ok(Self {
+                    status: enums::AttemptStatus::from(response),
+                    response: Ok(types::PaymentsResponseData::TransactionResponse {
+                        resource_id,
+                        redirection_data: None,
+                        mandate_reference: None,
+                        connector_metadata: None,
+                        network_txn_id: None,
+                        connector_response_reference_id: None,
+                        incremental_authorization_allowed: None,
+                        charge_id: None,
+                    }),
+                    ..item.data
+                })
+            }
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize)]
-pub struct RefundResponse {
-    id: String,
-    status: RefundStatus,
+impl TryFrom<&DatatransRouterData<&types::PaymentsCaptureRouterData>>
+    for DataPaymentCaptureRequest
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: &DatatransRouterData<&types::PaymentsCaptureRouterData>,
+    ) -> Result<Self, Self::Error> {
+        let amount = item.amount;
+        let currency = item.router_data.request.currency.to_string();
+        let refno = item.router_data.connector_request_reference_id.clone();
```

Comment:
Directly populate these values in the struct below instead of declaring variables.